Vision/src/V1/BatchOperationMetadata_State.php

```diff
@@ -0,0 +1,16 @@
+<?php
+# Generated by the protocol buffer compiler.  DO NOT EDIT!
+# source: google/cloud/vision/v1/product_search_service.proto
+
+namespace Google\Cloud\Vision\V1;
+
+if (false) {
+    /**
+     * This class is deprecated. Use Google\Cloud\Vision\V1\BatchOperationMetadata\State instead.
+     * @deprecated
+     */
+    class BatchOperationMetadata_State {}
```

Comment:
I'd love to see us not introduce classes in an already deprecated state. Have we discussed a plan for removing or being able to toggle these from the generator yet?

Reply:
This is tricky to do that in the generator because generator doesn't know which files are present in the latest release. If we really want to remove them, maybe we can remove it with the synth script?

Reply:
That's a good point. The synth script seems like the way to go, however - an additional item of note that I didn't originally take into account is that we would also need to update type checks to reference the new namespace instead of the old [(example)](https://github.com/googleapis/google-cloud-php/pull/1385/files#diff-890d1a8b719acb65c979781559631fc8R80). I'm not clear at the moment on all the implications that may be involved there, so it may be best to move forward as is for now and follow up removing the deprecated files later (which is a compromise I'm okay with as Vision is still beta).

WDYT?